torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -1318,7 +1388,7 @@ class TORCH_CUDA_CU_API ForLoop final : public Expr {
 //!
 class TORCH_CUDA_CU_API IfThenElse final : public Expr {
  public:
-  explicit IfThenElse(Passkey passkey, Bool* cond);
+  explicit IfThenElse(Passkey passkey, Predicate* cond);
```

Comment:
This seems to make the semantics of `IfThenElse` a little more specific and restrictive. `Predicate` is a very specific type of boolean values, representing predication of an expression, not just a plain bool. Making `IfThenElse` to be constructed with `Predicate` rather than `Bool` means that `IfThenElse` is only used for expression predication. While that's fine since that's the only usage of `IfThenElse`, and I expect that would be the case for the foreseeable future, we should document the intended usage model of `IfThenElse` here.

Reply:
The `PredicateType::Value` is used if you directly need a `Bool` value.

Reply:
Hmm, wouldn't it make sense to validate the predicate type is indeed `Value` when `Predicate::value()` is caled? Isn't it unsafe to use `Predicate::value()` when its type is not `Value`?

Reply:
I suppose we could enforce that invariant. I just check if value is not a nullptr. 

I used PredicateType::Value if I needed to manually generate a predicate from an existing Bool value.